### ISSUE TITLE
`Url.parse(...)` with `:` throws exception

### ISSUE THREAD

**Issue by jnatten** [lemonlabsuk/scala-uri#399]
**Describe the bug**
After upgrading to version 3.6.0 `Url.parse` (And other `.parse` methods) doesn't work with colon-characters (`:`).
I believe `:` is a valid url-character and should work. Seems to work as expected in `3.5.0`

**To Reproduce**
```scala
import io.lemonlabs.uri.Url

Url.parse("/this/works/fine")   // Works as expected
Url.parse("/this:1/does/not")   // Throws exception: `io.lemonlabs.uri.parsing.UriParsingException: Invalid URL could not be parsed. Error(5,NonEmptyList(EndOfString(5,16)))`
```

**Comment by theon** [comment]
Thank you both for raising this; this was an unintentional behaviour change between `3.5.0` and `3.6.0`; apologies. This has been fixed in `4.0.0-M5` and I am planning to release `4.0.0` in the next day or so.

**Comment by theon** [comment]
`4.0.0` has now been released. Thanks again both for raising this; much appreciated 🙇
